Userland/Libraries/LibSymbolication/Symbolication.cpp

```diff
@@ -61,24 +61,41 @@ Optional<FlatPtr> kernel_base()
 
 Optional<Symbol> symbolicate(String const& path, FlatPtr address)
 {
-    if (!s_cache.contains(path)) {
-        auto mapped_file = MappedFile::map(path);
-        if (mapped_file.is_error()) {
-            dbgln("Failed to map {}: {}", path, mapped_file.error().string());
+    String full_path = path;
+    if (!path.starts_with('/')) {
+        const char* search_paths[] = { "/usr/lib/{}", "/usr/local/lib/{}" };
+        bool found = false;
+        for (auto& search_path : search_paths) {
+            full_path = String::formatted(search_path, path);
```

Comment:
Generating paths using formatting like this is a bit awkward, how about just storing the the directories instead and using `LexicalPath::join(lib_directory, relative_path);`?

Reply:
Thanks @IdanHo for the pointers. I'm fairly new to the project so I'm still making my way through the standard library that's available. What I put here I based on https://github.com/SerenityOS/serenity/blob/683bf558e441fc3c0e448ca8807541e754adaa03/Userland/Libraries/LibELF/DynamicLinker.cpp#L112-L114

I looked into your suggestions (using `StringView`s + `LexicalPath::join`. I'll do that now, but I also realised that `LexicalPath::join` could probably accept `StringView`s instead of `String const&`s (otherwise we're creating `String`s for the `StringView`s anyway)? That improvement could be handled separately though.